contracts/bid/src/map.rs

```diff
@@ -43,13 +40,13 @@ where
     pub fn get(
         &self,
         pk: PublicKey,
-    ) -> Result<Option<impl Deref<Target = u64> + '_>, S::Error> {
-        self.0.get(&pk.to_bytes())
+    ) -> Result<Option<impl Deref<Target = u64> + '_>, CanonError> {
+        Ok(self.0.get(&pk.to_bytes())?)
     }
 
     /// Remove an entry from the tree. It will return `Ok(Some(u64))` in case
     /// the key exists and `Ok(None)` otherways.
-    pub fn remove(&mut self, pk: PublicKey) -> Result<Option<u64>, S::Error> {
-        self.0.remove(&pk.to_bytes())
+    pub fn remove(&mut self, pk: PublicKey) -> Result<Option<u64>, CanonError> {
+        Ok(self.0.remove(&pk.to_bytes())?)
```

Comment:
This change is not needed, you can remove the `Ok` and the `?` as it was before (the hamt's `remove` has the same return value of your `remove`).